From a914459d7b634ebb255b944ef02b1357a76c353d Mon Sep 17 00:00:00 2001
From: Grzegorz Jaszczyk <jaz@semihalf.com>
Date: Wed, 17 Feb 2016 13:04:13 +0100
Subject: [PATCH 0076/1345] ARM: mvebu: a382-db: align dts to LKv44
 conventions

commit  1b3dda481647d98089b19c33d6f657c690772715 from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

- use appropriate phy-mode (the same as for a388-db board)
- use correct node name for mdio
- align the spi-flash properties
- use mainline nand controller 'armada370-nand' instead of deprecated
hal-based and align properties basing on flash node of armada-388-db.dts

Change-Id: Ie314bd114818b7b35fb91cbc7b02ddba3dba2d6b
Signed-off-by: Grzegorz Jaszczyk <jaz@semihalf.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/27703
Reviewed-by: Lior Amsalem <alior@marvell.com>
Tested-by: Lior Amsalem <alior@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/armada-382-db.dts |   34 ++++++++++++++++------------------
 1 file changed, 16 insertions(+), 18 deletions(-)

diff --git a/arch/arm/boot/dts/armada-382-db.dts b/arch/arm/boot/dts/armada-382-db.dts
index b6626aa..af6b5eb 100644
--- a/arch/arm/boot/dts/armada-382-db.dts
+++ b/arch/arm/boot/dts/armada-382-db.dts
@@ -33,13 +33,13 @@
 			ethernet@70000 {
 				status = "okay";
 				phy = <&phy0>;
-				phy-mode = "rgmii";
+				phy-mode = "rgmii-id";
 			};
 
 			ethernet@30000 {
 				status = "okay";
 				phy = <&phy1>;
-				phy-mode = "rgmii";
+				phy-mode = "rgmii-id";
 			};
 
 			i2c0: i2c@11000 {
@@ -52,7 +52,7 @@
 				clock-frequency = <100000>;
 			};
 
-			mdio {
+			mdio@72004 {
 				phy0: ethernet-phy@0 {
 					reg = <0>;
 				};
@@ -82,8 +82,8 @@
 
 				spi-flash@0 {
 					#address-cells = <1>;
-					#size-cells = <1>;
-					compatible = "w25q32";
+					#size-cells = <0>;
+					compatible = "w25q32", "jedec,spi-nor";
 					reg = <0>; /* Chip select 0 */
 					spi-max-frequency = <108000000>;
 				};
@@ -99,30 +99,28 @@
 			 * unnessesary devices according to board
 			 * boot-up configuration.
 			 */
-			nfc: nand@d0000 {
+			flash@d0000 {
 				status = "okay";
-				#address-cells = <1>;
-				#size-cells = <1>;
-
-				nfc,nfc-mode  = "normal";	/* normal or ganged */
-				nfc,nfc-dma   = <0>;		/* 0 for no, 1 for dma */
-				nfc,nfc-width = <8>;
-				nfc,ecc-type  = <1>;		/* 4 bit */
-				nfc,num-cs    = <1>;
-
-				mtd0@00000000 {
+				num-cs = <1>;
+				marvell,nand-keep-config;
+				marvell,nand-enable-arbiter;
+				nand-on-flash-bbt;
+				nand-ecc-strength = <4>;
+				nand-ecc-step-size = <512>;
+
+				partition@0 {
 					label = "U-Boot";
 					reg = <0x00000000 0x00600000>;
 					read-only;
 				};
 
-				mtd1@00080000 {
+				partition@600000 {
 					label = "uImage";
 					reg = <0x00600000 0x00400000>;
 					read-only;
 				};
 
-				mtd2@00140000 {
+				partition@a000000 {
 					label = "Root";
 					reg = <0x00a00000 0x3f600000>;
 				};
-- 
1.7.9.5

